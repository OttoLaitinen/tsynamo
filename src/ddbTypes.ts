export type PartitionKey<T> = T & { _PK: true };

export type SortKey<T> = T & { _SK: true };
