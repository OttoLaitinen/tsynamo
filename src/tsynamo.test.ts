import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
import { DynamoDBDocumentClient } from "@aws-sdk/lib-dynamodb";
import { beforeAll, describe, expect, it, vi } from "vitest";
import { PartitionKey, SortKey } from "./ddbTypes";
import { Tsynamo } from "./index";

interface DDB {
  myTable: {
    userId: PartitionKey<string>;
    timestamp: SortKey<string>;
    somethingElse: string;
  };
}

describe("tsynamo", () => {
  const ddbClient = DynamoDBDocumentClient.from(new DynamoDBClient({}));
  const mockSend = vi.fn();

  beforeAll(() => {
    vi.spyOn(ddbClient, "send").mockImplementation(mockSend);
    mockSend.mockReturnValue(Promise.resolve({}));
  });

  it("sends a basic get command", async () => {
    const tsynamoClient = new Tsynamo<DDB>({
      ddbClient,
    });

    await tsynamoClient
      .getItemFrom("myTable")
      .keys({
        userId: "123",
        timestamp: "123",
      })
      .consistentRead(true)
      .attributes(["somethingElse"])
      .execute();

    expect(mockSend).toMatchSnapshot();
  });
});
