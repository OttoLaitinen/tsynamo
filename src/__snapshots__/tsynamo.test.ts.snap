// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tsynamo > sends a basic get command 1`] = `
[MockFunction spy] {
  "calls": [
    [
      GetCommand {
        "clientCommand": GetItemCommand {
          "deserialize": [Function],
          "input": {
            "AttributesToGet": [
              "somethingElse",
            ],
            "ConsistentRead": true,
            "Key": {
              "timestamp": "123",
              "userId": "123",
            },
            "TableName": "myTable",
          },
          "middlewareStack": {
            "add": [Function],
            "addRelativeTo": [Function],
            "applyToStack": [Function],
            "clone": [Function],
            "concat": [Function],
            "identify": [Function],
            "identifyOnResolve": [Function],
            "remove": [Function],
            "removeByTag": [Function],
            "resolve": [Function],
            "use": [Function],
          },
          "serialize": [Function],
        },
        "input": {
          "AttributesToGet": [
            "somethingElse",
          ],
          "ConsistentRead": true,
          "Key": {
            "timestamp": "123",
            "userId": "123",
          },
          "TableName": "myTable",
        },
        "inputKeyNodes": {
          "Key": {},
        },
        "middlewareStack": {
          "add": [Function],
          "addRelativeTo": [Function],
          "applyToStack": [Function],
          "clone": [Function],
          "concat": [Function],
          "identify": [Function],
          "identifyOnResolve": [Function],
          "remove": [Function],
          "removeByTag": [Function],
          "resolve": [Function],
          "use": [Function],
        },
        "outputKeyNodes": {
          "Item": {},
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": {},
    },
  ],
}
`;
